docs/arch.md

```diff
@@ -259,17 +256,32 @@ Such a client is used in [GatherMachineSet](pkg/gather/clusterconfig/clusterconf
 
 
 ## Gathering the data
-When the `periodic.go` calls method Gather of interface Gatherer, it actually calls `pkg/gather/clusterconfig/clusterconfig.go`.
-The function Gather calls one by one functions to gather (collect results of kubernetes Api call). Each `GatherXX` method is returning record object,
-which is a list of records with name (how file will be called in archive) and actual data. Actual data (any struct) has to implement Marshalable interface,
-which requires to have Marshall method.
 
-The structure of Gather calls is created inside the from `pkg/gather/clusterconfig` from Gather method. The method creates the list of gathering functions
-and passes them to record.Collect method in `pkg/record/interface.go`. The Collect method works in a way that any error returned from a gathering function
-is stored, but all next functions are still called, unless the parent context returns an Error (for example on timeout).
+### clusterconfig
+When the `periodic.go` calls method Gather of the `clusterconfig` Gatherer, it's handled [here](https://github.com/openshift/insights-operator/blob/master/pkg/gather/clusterconfig/0_gatherer.go#L99).
+
+The clusterconfig Gatherer starts each gather-function in its own separate goroutine with a dedicated channel to send back their results.
+Each gather-function is its own separate entity, each creates their own clients using the configs present in the `Gatherer` object that was passed down as parameter.
+We further cut the gather-functions into to two main parts:
```

Comment:
`We further divided the gather-functions into two main parts: ` sounds a bit better to me.